extensions/devservices/keycloak/runtime/src/main/java/io/quarkus/devservices/keycloak/runtime/KeycloakDevServicesRecorder.java

```diff
@@ -0,0 +1,106 @@
+package io.quarkus.devservices.keycloak.runtime;
+
+import java.util.function.Supplier;
+
+import org.jboss.logging.Logger;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+
+import io.quarkus.devconsole.runtime.spi.DevConsolePostHandler;
+import io.quarkus.runtime.annotations.Recorder;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.mutiny.ext.web.client.WebClient;
+
+@Recorder
+public class KeycloakDevServicesRecorder {
+    private static final Logger LOG = Logger.getLogger(KeycloakDevServicesRecorder.class);
+
+    public Supplier<String> getProviderName() {
+        return new Supplier<String>() {
+
+            @Override
+            public String get() {
+                return "KEYCLOAK";
+            }
+        };
+    }
+
+    public Supplier<String> getProviderUrlBase() {
+        return new Supplier<String>() {
+
+            @Override
+            public String get() {
+                return "io.quarkus" + "." + "quarkus-devservices-keycloak";
+            }
+        };
+    }
+
+    public Handler<RoutingContext> testService(Supplier<Vertx> vertx) {
+        return new DevConsolePostHandler() {
```

Comment:
This handler should be in the deployment side, not in the runtime.

Reply:
@stuartwdouglas, @cescoffier explained how to get a `Vertx` instance in the deployment, but for now I'm going to focus on replacing the `keycloak-authorization` code with Vert.x code - I already updated the token acquisition code and next I'll have to try and do the realm creation with Vert.x Web Client - hopefully that should be simple since Keycloak `RealmRepresentation` should be convertable to JSON using a Jackson mapper - it it works out then it will be possible to avoid the runtime updates; will look into it after PTO